core/src/main/scala/org/apache/spark/sql/delta/commands/OptimizeTableCommand.scala

```diff
@@ -164,18 +167,22 @@ class OptimizeExecutor(
 
   def optimize(): Seq[Row] = {
     recordDeltaOperation(txn.deltaLog, "delta.optimize") {
-      val minFileSize = sparkSession.sessionState.conf.getConf(
-        DeltaSQLConf.DELTA_OPTIMIZE_MIN_FILE_SIZE)
       val maxFileSize = sparkSession.sessionState.conf.getConf(
         DeltaSQLConf.DELTA_OPTIMIZE_MAX_FILE_SIZE)
-      require(minFileSize > 0, "minFileSize must be > 0")
       require(maxFileSize > 0, "maxFileSize must be > 0")
-
+      val (minFileSize, maxDeletedRowsRatio) = if (isPurge) {
```

Comment:
How about create a OptimizeConf and use it to pass the minFileSize and other required info? Can be done as part of the followup PRs.

Reply:
Done in this PR!